niceman/interface/diff.py

```diff
@@ -0,0 +1,150 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil; coding: utf-8 -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the niceman package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Report if a specification satisfies the requirements in another specification
+"""
+
+import sys
+import time
+
+from .base import Interface
+from ..support.constraints import EnsureStr
+from ..support.exceptions import InsufficientArgumentsError
+from ..support.param import Parameter
+
+__docformat__ = 'restructuredtext'
+
+from logging import getLogger
+lgr = getLogger('niceman.api.retrace')
+
+class MultipleDistributionsError(Exception):
+
+    """multiple distributions of a given type found"""
+
+    def __init__(self, cls):
+        self.cls = cls
+
+class Diff(Interface):
+    """Report if a specification satisfies the requirements in another 
+    specification
+
+    Examples
+    --------
+
+      $ niceman diff --env environment.yml --req required_environment.yml
+
+    """
+
+    _params_ = dict(
+        env=Parameter(
+            args=("--env",),
+            doc="NICEMAN environment specification", 
+            metavar='env',
+            constraints=EnsureStr()),
+        req=Parameter(
+            args=("--req",),
+            metavar="req",
+            doc="NICEMAN requirements", 
+            constraints=EnsureStr())
+    )
+
+    @staticmethod
+    def __call__(env, req):
+
+        if not env:
+            raise InsufficientArgumentsError("env undefined")
+
+        if not req:
+            raise InsufficientArgumentsError("req undefined")
```

Comment:
when you refactor it to be a list -- just verify that `len(specs)==2`. also make a unittest to trigger that check